File: Ruby/tests/problem071_test.rb
require 'minitest/autorun'
require_relative '../problem071'

class Problem071Test < Minitest::Test
  def test_rand5_has_an_even_chance_of_picking_each_elem
    number_space = (1..5000).map { |n| Problem071.rand5 }
    nums = number_space.uniq
    frequency = nums.map { |n| number_space.count { |x| n == x } }
    limit = 5000 / nums.length

    assert(frequency.all? { |n| n <= limit * 1.05 && n >= limit * 0.95 })
  end
end

File: JavaScript/tests/problem071_test.js
// Sample test file
const chai = require('chai')
const assert = chai.assert
const problem071 = require('../problem071');

// Helper
const getNumberSpace = (size) => {
  let out = []
  for (let i = 0; i < size; i++) {
    out.push(problem071.rand5())
  }
  return out
};

describe('Problem 071', function(){
  it("should have an equal chance of picking all numbers", function() {
    numberSpace = getNumberSpace(5000)
    uniqueNums = [...new Set(numberSpace)]
    frequency = uniqueNums.map((n) => numberSpace.filter((x) => n == x).length)
    limit = 1000

    assert.isTrue(frequency.every((n) => n <= limit * 1.05 && n >= limit * 0.95))
  });
});
